tests/Unit/VersionSwitcherTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Tests\Feature;
+
+use Tests\TestCase;
+use Illuminate\Http\Request;
+use App\Http\Middleware\VersionSwitcher;
+use App\Exceptions\AcceptHeaderWrongFormatException;
+
+class VersionSwitcherTest extends TestCase
+{
+    // Simulating route defined in the route definition of web.php.
+    protected $configuredRoute = [
+        1,
+        [
+            'uses' => 'App\Http\Controllers\ListController@get',
+            'middleware' => [
+                'auth',
+                'version-switcher',
+            ],
+        ],
+        [
+            'listId' => 'default',
+        ]
+    ];
+
+    /**
+     * @dataProvider versionRoutes
+     */
+    public function testThatMiddlewareSwapsControllerPathDependingOnVersion(string $version, array $expectedRoute): void
+    {
+        $request = $this->getMockBuilder(Request::class)
```

Comment:
Personally, I prefer using `$this->prophesize()`.

Reply:
And I prefer Mockery :)
I tried Mockery but I ran into an error becuase __clone is callled on the object somewhere in the flow and Mockery could not handle that. Ended up with using phpunits mock system.
Prophesize is nice to, agreed.